src/macros.rs

```diff
@@ -27,13 +27,18 @@ macro_rules! include_spv {
 /// The definition of this type is unstable.
 pub type Spirv = Align4<[u8]>;
 
-impl std::ops::Deref for Spirv {
-    type Target = [u32];
-    fn deref(&self) -> &[u32] {
-        #[allow(clippy::cast_ptr_alignment)]
-        unsafe {
-            std::slice::from_raw_parts(self.0.as_ptr() as *const u32, self.0.len() / 4)
+impl Spirv {
+    pub fn deref(&self) -> Vec<u32> {
+        let mut out = Vec::with_capacity(self.0.len() / 4);
+        for i in 0..self.0.len() / 4 {
+            let mut tmp: u32 = 0;
+            tmp += self.0[4 * i] as u32;
+            tmp += (self.0[4 * i + 1] as u32) << 8;
+            tmp += (self.0[4 * i + 2] as u32) << 16;
+            tmp += (self.0[4 * i + 3] as u32) << 24;
```

Comment:
This will have different behaviour compared to the original on big endian systems.